pkg/cmd/admin/groups/sync/syncerror/handler.go

```diff
@@ -0,0 +1,93 @@
+package syncerror
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+)
+
+// Handler knows how to handle errors
+type Handler interface {
+	// HandleError processess an error without mutating it. If the error is determined to be fatal,
+	// a non-nil error should be returned.
+	HandleError(err error) (handled bool, fatalError error)
+}
+
+func NewCompoundHandler(handlers ...Handler) Handler {
+	return &compoundHandler{handlers: handlers}
+}
+
+// compoundHandler chains other error handlers
+type compoundHandler struct {
+	handlers []Handler
+}
+
+// HandleError asks each of the handlers to handle the error. If any handler decides the error is fatal or handles it,
+// the chain is broken.
+func (h *compoundHandler) HandleError(err error) (bool, error) {
+	for _, handler := range h.handlers {
+		handled, handleErr := handler.HandleError(err)
+		if handled || handleErr != nil {
+			return handled, handleErr
+		}
+	}
+
+	// if nobody could handle the error, we consider it fatal
+	return false, err
+}
+
+func NewMemberLookupOutOfBoundsSuppressor(err io.Writer) Handler {
+	return &memberLookupOutOfBoundsSuppresor{err: err}
+}
+
+// memberLookupOutOfBoundsSuppresor suppresses member lookup errors caused by a search trying to go out of the base DN bounds
+type memberLookupOutOfBoundsSuppresor struct {
+	// err determines where a log message will be printed
+	err io.Writer
+}
+
+// HandleError suppresses member lookup errors caused by out-of-bounds queries,
+func (h *memberLookupOutOfBoundsSuppresor) HandleError(err error) (bool, error) {
+	memberLookupError, isMemberLookupError := err.(*memberLookupError)
+	if !isMemberLookupError {
+		return false, nil
+	}
+
+	if !ldaputil.IsQueryOutOfBoundsError(memberLookupError.causedBy) {
```

Comment:
switch to positive, print and `return true,nil`, and let the default case at the end of the function return `false, nil`